hivemq-extension-sdk/src/main/java/com/hivemq/extension/sdk/api/packets/puback/ModifiablePubackPacket.java

```diff
@@ -0,0 +1,49 @@
+package com.hivemq.extension.sdk.api.packets.puback;
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extension.sdk.api.annotations.Nullable;
+import com.hivemq.extension.sdk.api.packets.general.ModifiableUserProperties;
+import com.hivemq.extension.sdk.api.packets.general.UserProperties;
+import com.hivemq.extension.sdk.api.packets.publish.AckReasonCode;
+
+/**
+ * A {@link PubackPacket} that can be modified before it is sent to the client.
+ *
+ * @author Yannick Weber
+ */
+public interface ModifiablePubackPacket extends PubackPacket {
+
+    /**
+     * Set a {@link AckReasonCode} to the PUBACK packet.
+     * <p>
+     * Switching from successful to unsuccessful and vice versa is not supported.
+     *
+     * @param reasonCode The reason code to set.
+     * @throws NullPointerException  If reason code is <null>.
+     * @throws IllegalStateException If switching from successful reason code to unsuccessful reason code or vice versa.
+     * @see AckReasonCode How reason codes are translated from MQTT 5 to MQTT 3.
+     */
+    void setReasonCode(final @NotNull AckReasonCode reasonCode);
```

Comment:
I dont like AckReasonCode. That sounds bad. Perhapps make it abstract and add an PubackReasonCode?